protocols/grpc/agent.proto

```diff
@@ -170,29 +174,29 @@ message Route {
 	string device = 3;
 }
 
-enum UpdateType {
-	None = 0;
-	AddIP = 1;
-	RemoveIP = 2;
-	Name = 4;
-	MTU = 8;
-}
-
 message AddInterfaceRequest {
-	Interface interface = 1;
+	string interface_id = 1;
+	Interface interface = 2;
 }
 
 message RemoveInterfaceRequest {
-	string name = 1;
+	string interface_id = 1;
 }
 
+// interface_id MUST NOT empty
+// all the other fields should be updated when it is set
 message UpdateInterfaceRequest {
-	UpdateType type = 1;
-	Interface interface = 2;
+	string interface_id = 1;
+	string device = 2;
+	string name = 3;
+	repeated IPAddress remove_addresses = 4;
+	repeated IPAddress add_addresses = 5;
+	uint64 mtu = 6;
+	string hwAddr = 7;
 }
 
 message RouteRequest {
-	Route route = 1;
+	repeated Route routes = 1;
```

Comment:
Why do we change that one as well?

Reply:
a list of route is more likely to be added rather than a single one.

Reply:
Same discussion on `declarative API` vs `imperative API`